src/components/application_manager/src/hmi_state.cc

```diff
@@ -49,53 +49,54 @@ HmiState::HmiState(utils::SharedPtr<Application> app,
 
 HmiState::HmiState(utils::SharedPtr<Application> app,
                    const ApplicationManager& app_mngr)
-    : app_(app)
+    : app_id_(app->app_id())
     , state_id_(STATE_ID_REGULAR)
     , app_mngr_(app_mngr)
     , hmi_level_(mobile_apis::HMILevel::INVALID_ENUM)
     , audio_streaming_state_(mobile_apis::AudioStreamingState::INVALID_ENUM)
     , system_context_(mobile_apis::SystemContext::INVALID_ENUM) {}
 
-DEPRECATED HmiState::HmiState(uint32_t app_id,
-                              const ApplicationManager& app_mngr,
-                              StateID state_id)
-    : state_id_(state_id)
+HmiState::HmiState(uint32_t app_id,
+                   const ApplicationManager& app_mngr,
+                   StateID state_id)
+    : app_id_(app_id)
+    , state_id_(state_id)
     , app_mngr_(app_mngr)
     , hmi_level_(mobile_apis::HMILevel::INVALID_ENUM)
     , audio_streaming_state_(mobile_apis::AudioStreamingState::INVALID_ENUM)
-    , system_context_(mobile_apis::SystemContext::INVALID_ENUM) {
-  app_ = app_mngr_.application(app_id);
-}
+    , system_context_(mobile_apis::SystemContext::INVALID_ENUM) {}
 
-DEPRECATED HmiState::HmiState(uint32_t app_id,
-                              const ApplicationManager& app_mngr)
-    : state_id_(STATE_ID_REGULAR)
+HmiState::HmiState(uint32_t app_id, const ApplicationManager& app_mngr)
+    : app_id_(app_id)
+    , state_id_(STATE_ID_REGULAR)
     , app_mngr_(app_mngr)
     , hmi_level_(mobile_apis::HMILevel::INVALID_ENUM)
     , audio_streaming_state_(mobile_apis::AudioStreamingState::INVALID_ENUM)
-    , system_context_(mobile_apis::SystemContext::INVALID_ENUM) {
-  app_ = app_mngr_.application(app_id);
-}
+    , system_context_(mobile_apis::SystemContext::INVALID_ENUM) {}
 
 void HmiState::set_parent(HmiStatePtr parent) {
   DCHECK_OR_RETURN_VOID(parent);
   parent_ = parent;
 }
 
 bool HmiState::is_navi_app() const {
-  return app_->is_navi();
+  const ApplicationSharedPtr app = app_mngr_.application(app_id_);
```

Comment:
@shoamano83 according to my comment above, propose to use `application_by_hmi_app()` instead

Reply:
Updated with commit 14270c9b11d5ac824f1d615d8b2f36277d573b53.